sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -600,14 +600,23 @@ public Symbol visitArrayComparisonExpression(ArrayComparisonExpression node, Exp
                 throw new UnsupportedFeatureException("ALL is not supported");
             }
 
-            Symbol leftSymbol = process(node.getLeft(), context);
-            Symbol arraySymbol = process(node.getRight(), context);
+            Expression left = node.getLeft();
+            Expression right = node.getRight();
+
+            if (right instanceof SubqueryExpression) {
+                context.registerSubqueryExpression((SubqueryExpression) right);
```

Comment:
This doesn't work because we want to apply the `single_value` function iff the `SubqueryExpression` is **not** part of an array expression. Above, we would have to unwrap the single_value function again.